### ISSUE TITLE
Multithreading code with NEON is slower than just NEON

### ISSUE THREAD

**Issue by ihhub** [ihhub/penguinV#147]
I ran performance tests on Raspberry Pi 3B and surprisedly many tests with multithreading (function pool) and NEON are slower than the normal code with NEON only.

There is an example of timings:
> [73/220] image_function_neon::Invert (2048x2048)... 3.951+/-1.928 ms
[74/220] image_function_neon::Invert (1024x1024)... 1.521+/-1.147 ms
[75/220] image_function_neon::Invert (512x512)... 0.2575+/-0.04695 ms
[76/220] image_function_neon::Invert (256x256)... 0.04703+/-0.003153 ms
[77/220] function_pool::Invert (2048x2048)... 5.906+/-1.233 ms
[78/220] function_pool::Invert (1024x1024)... 1.607+/-1.079 ms
[79/220] function_pool::Invert (512x512)... 0.5198+/-0.8268 ms
[80/220] function_pool::Invert (256x256)... 0.1467+/-0.05831 ms
[81/220] image_function::Invert (2048x2048)... 22.26+/-25.71 ms
[82/220] image_function::Invert (1024x1024)... 3.634+/-0.1714 ms
[83/220] image_function::Invert (512x512)... 0.9299+/-0.1116 ms
[84/220] image_function::Invert (256x256)... 0.2293+/-0.02301 ms

We need to figure our whether NEON code is actually called within function pool and if yes then what's the problem behind.

**Comment by 0x72D0** [comment]
It shouldn't be too long to test, I'll test it

**Comment by ihhub** [comment]
I made a deep research and I found that on my machine (which is CISC architecture) I have almost the same performance for **Invert** (or many more functions). Things which I found:
1. My PC as well as Raspberry Pi has only one physical piece of RAM. In this case a parallel access to RAM by multiple threads wouldn't be possible leading to memory restriction in speed
2. Between 1024 and 2048 the gap in performance in about 16 times. It is due to CPU caching which I briefly mentioned in function pool class
3. This function is so simple so memory operation is prevalent compare to CPU work so we don't wee any performance improvement for multithreaded version.
